### ISSUE TITLE
Kamstrup401 working?

### ISSUE THREAD

**Issue by madradrakie** [matthijsberk/M401-irreader#2]
Hello Matthijs. I was wondering if you are able to the data from the kampstrup 401 meter? If you managed it, can you help me implement your solution / code?

Thanks in advance.

https://github.com/matthijsvisser/kamstrup-402-mqtt/issues/7#issuecomment-1289044269

**Comment by matthijsberk** [comment]
Yes, it is working at the moment. Unfortunately the code is really sloppy and I don't have time to optimise it further at the moment. If you want to run it, you need to download all the files and fill in the details and USB port in the config.yaml file. You also need to fill in your USB port in the daemon.py file at row 80. If you run daemon.py after the changes it should work.

You can also test run it by using [kamstrupp.python](https://github.com/matthijsberk/M401-irreader/blob/main/kamstrupp.python) and only fill in the correct USB port of the infrared sensor. If you run this python script it should print the values it gets from you meter.

**Comment by madradrakie** [comment]
thanks matthijs. can i follow the following steps to install the script? (step 3)
https://www.pieterbrinkman.com/2022/02/01/make-your-city-heating-stadsverwarming-smart-and-connect-it-home-assistant-energy-dashboard/

apt-get install git
git clone https://github.com/matthijsberk/M401-irreader.git

**Comment by madradrakie** [comment]
i think i got it almost working, the reader is giving me values ;-)
which poll interval do you use? For the 402 it is 30 minutes, did you use 180 minutes?

**Comment by matthijsberk** [comment]
Yep, thats correct. You can use the steps. I have a poll interval of 60 minutes (otherwise the battery will be drained too much in my opinion.). And remember, the code is sloppy so it is not the most efficient or prettiest code, but works for now.

**Comment by madradrakie** [comment]
thanks @matthijsberk. it is working ;-) (Y)
